### PR TITLE
FEA-2421: Fixed Functions as Parameters

### LINKED ISSUE #86: Functions as Parameters Indexing Issues
Given the following situation, where a function, takes a function parameter, and that nested function contains a parameter

```dart
void func1(void Function(String) bar) { ... }
void func2(void Function(int) bar) { ... }
```

2 (potentially related) results will be indexed:

- both func1 and func2's parameter for `bar`'s `String`/`int` will have an occurrence location of:
    ```go
    []int32{ 0, -1, -1 }
    ```
    which errors out the `snapshot` scip cli action, and is just not correct

- both func1 and func2's parameter for `bar`'s `String`/`int` have a duplicate local symbol: `local 1`

----

Additional information:

scip-dart currently treats all parameters (except named) as `local x`. This is because only named parameters can be goto'd from consumers, all others are only declared locally.

This is not the case for scip-typescript where the following is outputted:
```ts
function func1(foo: String) {}
//             ^^^ func1().(foo)

let func2 = (foo: String) {}
//           ^^^ local 1
```

Named functions will always have their parameters indexed, where as if the function is anonymous (eg, the symbol for the function is not calculate-able) the parameter is treated as `local x`

This is not now `scip-dart` functions, where all parameters except named ones are considered `local x`. This is a minor different that shouldn't have an effect on indexing (as far as I'm aware that is). Potentially do some further digging around this to consider if we can continue to pursue this pattern of always local parameters

### PR DESCRIPTION
**Pull request by matthewnitschke-wk**
# [FEA-2421](https://jira.atl.workiva.net/browse/FEA-2421)
Issue Status

Closes #86

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: pubspec.lock [hunk 1]
```diff
@@ -7,21 +7,21 @@ packages:
       name: _fe_analyzer_shared
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "61.0.0"
+    version: "54.0.0"
```
Comment (matthewnitschke-wk) [inline pubspec.lock:10]: dependabot pub does not effect the pubspec.yaml directly, and when it does bump versions, it doesnt care about bumping past the `sdk:` field declared in the pubspec.yaml

For example: analyzer `5.13.0` requires dart `2.19`, but dependabot happily bumped the lockfile to an invalid version. [this](https://github.com/Workiva/scip-dart/pull/88/files#diff-dd4fbda47e51f1e35defb9275a9cd9c212ecde0b870cba89ddaaae65c5f3cd28R4) change is an effort to see what a different versioning strategy does to the resolution logic in dart

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
```
Comment (matthewnitschke-wk) [inline .vscode/launch.json:6]: This lets the debugger be used for indexing `staging-project`

File: .github/dependabot.yml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 version: 2
 updates:
-  - package-ecosystem: "pub"
-    directory: "/"
+  - package-ecosystem: pub
+    versioning-strategy: increase
```
Comment (matthewnitschke-wk) [inline .github/dependabot.yml:4]: see: https://github.com/Workiva/scip-dart/pull/88/files#r1308142900

**Review by corwinsheahan-wf — APPROVED** [review]

**Comment by matthewnitschke-wk** [comment]
QA +1

- [x] Generated snapshots are as expected

🚀 @Workiva/release-management-p 🚢

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**